src/titan_db_test.cc

```diff
@@ -252,10 +254,17 @@ class TitanDBTest : public testing::Test {
   }
 
   void CallGC() {
-    db_impl_->bg_gc_scheduled_++;
+    {
+      MutexLock l(&db_impl_->mutex_);
+      db_impl_->bg_gc_scheduled_++;
+    }
     db_impl_->BackgroundCallGC();
-    while (db_impl_->bg_gc_scheduled_)
-      ;
+    {
+      MutexLock l(&db_impl_->mutex_);
+      while (db_impl_->bg_gc_scheduled_) {
```

Comment:
mutex is held, `bg_gc_scheduled_` would not change

Reply:
cv_.Wait() implicitly unlock mutex. This is common pattern to use condition variables.

Reply:
got it